include/picongpu/fields/incidentField/profiles/ExpRampWithPrepulse.hpp

```diff
@@ -42,35 +43,30 @@ namespace picongpu
                      * @tparam T_Params user (SI) parameters
                      */
                     template<typename T_Params>
-                    struct ExpRampWithPrepulseUnitless : public T_Params
+                    struct ExpRampWithPrepulseUnitless : public BaseTransversalGaussianParamUnitless<T_Params>
                     {
+                        //! User SI parameters
                         using Params = T_Params;
 
-                        static constexpr float_X WAVE_LENGTH
-                            = static_cast<float_X>(Params::WAVE_LENGTH_SI / UNIT_LENGTH); // unit: meter
-                        static constexpr float_X PULSE_LENGTH
-                            = static_cast<float_X>(Params::PULSE_LENGTH_SI / UNIT_TIME); // unit: seconds (1 sigma)
+                        //! Base unitless parameters
+                        using Base = BaseTransversalGaussianParamUnitless<T_Params>;
+
+                        // unit: UNIT_TIME
                         static constexpr float_X LASER_NOFOCUS_CONSTANT
-                            = static_cast<float_X>(Params::LASER_NOFOCUS_CONSTANT_SI / UNIT_TIME); // unit: seconds
-                        static constexpr float_X AMPLITUDE
-                            = static_cast<float_X>(Params::AMPLITUDE_SI / UNIT_EFIELD); // unit: Volt /meter
-                        static constexpr float_X W0_AXIS_1
-                            = static_cast<float_X>(Params::W0_AXIS_1_SI / UNIT_LENGTH); // unit: meter
-                        static constexpr float_X W0_AXIS_2
-                            = static_cast<float_X>(Params::W0_AXIS_2_SI / UNIT_LENGTH); // unit: meter
+                            = static_cast<float_X>(Params::LASER_NOFOCUS_CONSTANT_SI / UNIT_TIME);
 
-                        static constexpr float_64 TIME_PREPULSE
-                            = static_cast<float_64>(Params::TIME_PREPULSE_SI / UNIT_TIME);
-                        static constexpr float_64 TIME_PEAKPULSE
-                            = static_cast<float_64>(Params::TIME_PEAKPULSE_SI / UNIT_TIME);
-                        static constexpr float_64 TIME_1 = static_cast<float_64>(Params::TIME_POINT_1_SI / UNIT_TIME);
-                        static constexpr float_64 TIME_2 = static_cast<float_64>(Params::TIME_POINT_2_SI / UNIT_TIME);
-                        static constexpr float_64 TIME_3 = static_cast<float_64>(Params::TIME_POINT_3_SI / UNIT_TIME);
+                        static constexpr float_X TIME_PREPULSE

```

Comment:
precision change from 64 to 32 bit.

Reply:
This was actually `float_64` in the original laser profile. I check how it's used, there is no requirement to have it as `float_64`, so changing to `float_X` appears fine.